code/modules/projectiles/projectile.dm

```diff
@@ -41,46 +50,24 @@
 	var/forcedodge = 0
 	// 1 to pass solid objects, 2 to pass solid turfs (results in bugs, bugs and tons of bugs)
 
-/obj/item/projectile/New()
-	permutated = list()
-	return ..()
+	var/proj_hit = 0
 
 /obj/item/projectile/proc/Range()
-	range--
-	if(range <= 0 && loc)
-		on_range()
+	if(range)
+		range--
+		if(range <= 0)
+			on_range()
 
 /obj/item/projectile/proc/on_range() //if we want there to be effects when they reach the end of their range
+	proj_hit = 1
 	qdel(src)
 
 /obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)
-	if(!isliving(target))
-		return 0
+	if(!isliving(target))	return 0

```

Comment:
put these back on the line below, rather than on the same line as the statement.